File: sdks/insuretech-go-sdk/pkg/models/claim_approval.go
package models

import (
	"time"
)

// ClaimApproval represents a claim_approval
type ClaimApproval struct {
	ApprovedAmount *Money `json:"approved_amount,omitempty"`
	ApprovalId string `json:"approval_id,omitempty"`
	ClaimId string `json:"claim_id,omitempty"`
	Decision *ApprovalDecision `json:"decision,omitempty"`
	Notes string `json:"notes,omitempty"`
	ApprovedAt time.Time `json:"approved_at,omitempty"`
	CreatedAt time.Time `json:"created_at,omitempty"`
	ApprovedCurrency string `json:"approved_currency,omitempty"`
	ApproverId string `json:"approver_id,omitempty"`
	ApproverRole string `json:"approver_role,omitempty"`
	ApprovalLevel int `json:"approval_level,omitempty"`
}

File: backend/insurance_engine/src/InsuranceEngine.Policy/Domain/Entities/ClaimApproval.cs
using System;
using InsuranceEngine.Policy.Domain.Enums;

namespace InsuranceEngine.Policy.Domain.Entities;

public class ClaimApproval
{
    public Guid Id { get; set; }
    public Guid ClaimId { get; set; }
    public Guid ApproverId { get; set; }
    public string ApproverRole { get; set; } = string.Empty;
    public int ApprovalLevel { get; set; }
    public ApprovalDecision Decision { get; set; }
    public long ApprovedAmount { get; set; }
    public string ApprovedCurrency { get; set; } = "BDT";
    public string? Notes { get; set; }
    public DateTime ApprovedAt { get; set; }
    public DateTime CreatedAt { get; set; }
}
